### ISSUE TITLE
The "LinkAssemblies" task failed unexpectedly in release mode.

### ISSUE THREAD

**Issue by elyfran** [TorbenK/TK.CustomMap#130]
C:\Program Files (x86)\MSBuild\Xamarin\Android\Xamarin.Android.Common.targets(5,5): Error MSB4018: The "LinkAssemblies" task failed unexpectedly.
Java.Interop.Tools.Diagnostics.XamarinAndroidException: error XA2006: Reference to metadata item 'System.Void Xamarin.Forms.Maps.Android.MapRenderer::OnElementChanged(Xamarin.Forms.Platform.Android.ElementChangedEventArgs`1<Xamarin.Forms.View>)' (defined in 'TK.CustomMap.Android, Version=1.5.0.0, Culture=neutral, PublicKeyToken=null') from 'TK.CustomMap.Android, Version=1.5.0.0, Culture=neutral, PublicKeyToken=null' could not be resolved. ---> Mono.Cecil.ResolutionException: Failed to resolve System.Void Xamarin.Forms.Maps.Android.MapRenderer::OnElementChanged(Xamarin.Forms.Platform.Android.ElementChangedEventArgs`1<Xamarin.Forms.View>)
   at Mono.Linker.Steps.MarkStep.MarkMethod(MethodReference reference)
   at Mono.Linker.Steps.MarkStep.MarkInstruction(Instruction instruction)
   at Mono.Linker.Steps.MarkStep.MarkMethodBody(MethodBody body)
   at Mono.Linker.Steps.MarkStep.ProcessMethod(MethodDefinition method)
   at Mono.Linker.Steps.MarkStep.ProcessQueue()
   at Mono.Linker.Steps.MarkStep.Process()
   at Mono.Linker.Steps.MarkStep.Process(LinkContext context)
   at Mono.Linker.Pipeline.Process(LinkContext context)
   at MonoDroid.Tuner.Linker.Process(LinkerOptions options, LinkContext& context)
   at Xamarin.Android.Tasks.LinkAssemblies.Execute()
   --- End of inner exception stack trace ---
   at Java.Interop.Tools.Diagnostics.Diagnostic.Error(Int32 code, Exception innerException, String message, Object[] args)
   at Xamarin.Android.Tasks.LinkAssemblies.Execute()
   at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute()
   at Microsoft.Build.BackEnd.TaskBuilder.<ExecuteInstantiatedTask>d__26.MoveNext() (MSB4018) (CpRAS.Droid)

**Comment by TorbenK** [comment]
Try clearing your project and make sure your Android SDK is up to date

**Comment by Troto** [comment]
@TorbenK I'm having the same problem. Any other suggestions? 

I'd really like to have a solution to this ASAP, your plugin is saving me a lot of work so I'd love to get it building.

**Comment by Troto** [comment]
@TorbenK Please could you look into this.

**Comment by TorbenK** [comment]
Your Android SDK is up to date? Did you clear the solution?

If that didn't work, can you compile using "Link SDK Assemblies only"?

I can maybe look at it this later today.

**Comment by Troto** [comment]
@TorbenK I have cleaned, rebuilt and restarted multiple times. 

I'm using platform 7.0 and that's at the latest version from the SDK manager.

I have been trying with "Link SDK Assemblies only" and it's failing. I can build with "None". I haven't even tried linking with SDK and User assemblies. I also tried adding TKCustomMap to the linked assemblies to skip but it didn't work.

It fails in Release build with "Link SDK Assemblies only".

Sorry to hassle you with a double comment but I'm hoping to release something soon and need to be able to make an APK smaller than 60MB lol
